apps/rewards/src/invite/inviteReward.service.ts

```diff
@@ -0,0 +1,171 @@
+import { KomenciLoggerService } from '@app/komenci-logger'
+import { networkConfig, NetworkConfig } from '@app/utils/config/network.config'
+import { isAccountConsideredVerified } from '@celo/base/lib'
+import { CeloContract, ContractKit } from '@celo/contractkit'
+import { Inject, Injectable } from '@nestjs/common'
+import { Cron, CronExpression } from '@nestjs/schedule'
+import { Raw } from 'typeorm'
+import { v4 as uuidv4 } from 'uuid'
+import { NotifiedBlock } from '../blocks/notifiedBlock.entity'
+import { NotifiedBlockRepository } from '../blocks/notifiedBlock.repository'
+import { InviteReward, RewardStatus } from './inviteReward.entity'
+import { InviteRewardRepository } from './inviteReward.repository'
+
+const WEEKLY_INVITE_LIMIT = 20
+
+@Injectable()
+export class InviteRewardService {
+  isRunning = false
+  komenciAddresses = []
+
+  constructor(
+    private readonly inviteRewardRepository: InviteRewardRepository,
+    private readonly notifiedBlockRepository: NotifiedBlockRepository,
+    private readonly contractKit: ContractKit,
+    @Inject(networkConfig.KEY)
+    private readonly networkCfg: NetworkConfig,
+    private readonly logger: KomenciLoggerService
+  ) {
+    this.komenciAddresses = this.networkCfg.relayers.map(
```

Comment:
We'll need to think through how to handle key rotations in the future. Very unlikely that we'll need it but good to have a game plan for how to handle it.

Reply:
If we do key rotations wouldn't we be able to redeploy the service? If we change the keys and deploy it this should be able to pick the new ones up, right? That was my understanding at least 😅

Reply:
Yeah, that's correct. Just thinking it might be tricky to provide guarantees when processing historical events since we would also need to track the old keys and know when the rotation occurred. Realistically we can just make sure that if we do a key rotation, we have rewarded everyone up to that point. Very unlikely key rotation would ever be needed.